provision/kubernetes/deploy.go

```diff
@@ -306,29 +307,67 @@ func extraRegisterCmds(a provision.App) string {
 	return fmt.Sprintf(`curl -sSL -m15 -XPOST -d"hostname=$(hostname)" -o/dev/null -H"Content-Type:application/x-www-form-urlencoded" -H"Authorization:bearer %s" %sapps/%s/units/register || true`, token, host, a.GetName())
 }
 
-func probeFromHC(hc provision.TsuruYamlHealthcheck, port int) (*apiv1.Probe, error) {
+type hcResult struct {
+	liveness     *apiv1.Probe
+	readiness    *apiv1.Probe
+	postStartCmd string
+}
+
+func probesFromHC(hc provision.TsuruYamlHealthcheck, port int) (hcResult, error) {
+	var result hcResult
 	if hc.Path == "" {
-		return nil, nil
-	}
-	scheme := hc.Scheme
-	if scheme == "" {
-		scheme = provision.DefaultHealthcheckScheme
-	}
-	scheme = strings.ToUpper(scheme)
-	method := strings.ToUpper(hc.Method)
-	if method != "" && method != "GET" {
-		return nil, errors.New("healthcheck: only GET method is supported in kubernetes provisioner")
-	}
-	return &apiv1.Probe{
+		return result, nil
+	}
+	if hc.Scheme == "" {
+		hc.Scheme = provision.DefaultHealthcheckScheme
+	}
+	hc.Method = strings.ToUpper(hc.Method)
+	if hc.Method == "" {
+		hc.Method = http.MethodGet
+	}
+	if hc.IntervalSeconds == 0 {
+		hc.IntervalSeconds = 10
+	}
+	if hc.TimeoutSeconds == 0 {
+		hc.TimeoutSeconds = 60
+	}
+	if !hc.UseInRouter {
+		url := fmt.Sprintf("%s://localhost:%d/%s", hc.Scheme, port, strings.TrimPrefix(hc.Path, "/"))
+		// Use postStart lifecycle hook executing curl in a while loop. We do
+		// this because we only want this healthcheck to be called once the
+		// container is started however we must still retry it on connection
+		// errors or after a HTTP error if AllowedFailures is > 1. This is the
+		// reason why we only increment the counter if curl returns status code
+		// 22 (response code >= 400 according to curl docs)
+		result.postStartCmd = fmt.Sprintf(`i=0
+while [ $i -lt %[1]d ] && ! curl -sSf -X%[2]s -o /dev/null -m %[3]d %[4]s; do [ $? -eq 22 ] && let i=i+1; sleep 3; done
```

Comment:
`sleep 3` for compatibility with docker-cluster?

Reply:
Yep, this was for compatibility as we have 3 seconds hardcoded in provision/docker. Using interval seconds here didn't seem right as the meaning is quite different from when it's used in the probes.